File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/Block/PostBlockDto.java
package com.salesianos.triana.BeElite.dto.Block;

import com.salesianos.triana.BeElite.dto.Set.PostSetDto;
import com.salesianos.triana.BeElite.exception.NotFoundException;
import com.salesianos.triana.BeElite.model.Block;
import com.salesianos.triana.BeElite.model.Composite_Ids.AthleteSessionId;
import com.salesianos.triana.BeElite.model.Composite_Ids.BlockId;
import com.salesianos.triana.BeElite.model.Composite_Ids.SessionId;
import com.salesianos.triana.BeElite.model.Session;
import com.salesianos.triana.BeElite.repository.SessionRepository;
import lombok.Builder;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;

@Builder
public record PostBlockDto(
        Long block_number,
        String movement,
        String block_instructions,
        Double rest_between_sets,
        List<PostSetDto> sets
) {

    public static PostBlockDto of(Block b){
        return PostBlockDto.builder()
                .block_number(b.getBlock_id().getBlock_number())
                .movement(b.getMovement())
                .block_instructions(b.getInstructions())
                .rest_between_sets(b.getRest_between_sets())
                .sets(b.getSets() != null ?
                        b.getSets().stream().map(PostSetDto::of).toList() :
                        List.of())
                .build();
    }
}

File: frontend/be_elite/lib/models/Session/post_session_dto/post_block_dto.dart
import 'post_set_dto.dart';

class PostBlockDto {
	int? blockNumber;
	String? movement;
	String? blockInstructions;
	int? restBetweenSets;
	List<PostSetDto>? sets;

	PostBlockDto({
		this.blockNumber, 
		this.movement, 
		this.blockInstructions, 
		this.restBetweenSets, 
		this.sets, 
	});

	factory PostBlockDto.fromJson(Map<String, dynamic> json) => PostBlockDto(
				blockNumber: json['block_number'] as int?,
				movement: json['movement'] as String?,
				blockInstructions: json['block_instructions'] as String?,
				restBetweenSets: json['rest_between_sets'] as int?,
				sets: (json['sets'] as List<dynamic>?)
						?.map((e) => PostSetDto.fromJson(e as Map<String, dynamic>))
						.toList(),
			);

	Map<String, dynamic> toJson() => {
				'block_number': blockNumber,
				'movement': movement,
				'block_instructions': blockInstructions,
				'rest_between_sets': restBetweenSets,
				'sets': sets?.map((e) => e.toJson()).toList(),
			};
}
